proposed/2021/DotnetNugetTrustedSigners.md

```diff
@@ -0,0 +1,196 @@
+
+USAGE:
+    dotnet nuget trusted-signer [OPTIONS] [COMMAND]
+
+COMMANDS:
+    list (default)
+    trust-package <name> <package>...
+    trust-source <name> [source-url]
+    trust-certificate <name> <fingerprint>
+    remove <name>
+    sync <name>
+```
+
+#### `dotnet nuget trusted-signers list`
+
+```
+Lists all the trusted signers in the configuration. This option will include all the certificates (with fingerprint and fingerprint algorithm) each signer has. If a certificate has a preceding [U], it means that certificate entry has allowUntrustedRoot set as true.
+
+USAGE:
+    dotnet nuget trusted-signers [OPTIONS] list
+
+OPTIONS:
+    -h, --help                          Prints help information.
+    -v, --verbosity <LEVEL>             Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+        --interactive                   Allows command to stop and wait for user input or action.
+        --configfile                    The NuGet configuration file. If specified, only the settings from this file will be used. If not specified, the hierarchy of configuration files from the current directory will be used.
+
+EXAMPLE:
+    $ dotnet nuget trusted-signers list
+    Registered trusted signers:
+
+    1.   nuget.org [repository]
+        Service Index: https://api.nuget.org/v3/index.json
+        Certificate fingerprint(s):
+            SHA256 - 0E5F38F57DC1BCC806D8494F4F90FBCEDD988B46760709CBEEC6F4219AA6157D
+
+    2.   microsoft [author]
+        Certificate fingerprint(s):
+            SHA256 - 3F9001EA83C560D712C24CF213C3D312CB3BFF51EE89435D3430BD06B5D0EECE
+            SHA256 - AA12DA22A49BCE7D5C1AE64CC1F3D892F150DA76140F210ABD2CBFFCA2C18A27
+
+    3.   myUntrustedAuthorSignature [author]
+        Certificate fingerprint(s):
+            [U] SHA256 - 518F9CF082C0872025EFB2587B6A6AB198208F63EA58DD54D2B9FF6735CA4434
+```
+
+#### `dotnet nuget trusted-signers sync`
+
+```
+Requests the latest list of certificates used in a currently trusted repository to update the the existing certificate list in the trusted signer.
+
+Note: This gesture will delete the current list of certificates and replace them with an up-to-date list from the repository.
+
+USAGE:
+    dotnet nuget trusted-signers sync [OPTIONS] <name>
+
+OPTIONS:
+    -h, --help                          Prints help information.
+    -v, --verbosity <LEVEL>             Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+        --interactive                   Allows command to stop and wait for user input or action.
+        --configfile                    The NuGet configuration file. If specified, only the settings from this file will be used. If not specified, the hierarchy of configuration files from the current directory will be used.
+```
+
+#### `dotnet nuget trusted-signers remove`
+
+```
+Removes any trusted signers that match the given name.
```

Comment:
According to the [code ](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/TrustedSignersCommand/TrustedSignersCommandRunner.cs#L52), there is a checking on duplicate names in trusted-signers section of config file, to guarantee that there is no duplicate name, among all trusted-signers (both author and repository signers). So it might be good to just specifying the name, without 'type'.